File: src/app/model/WheelRotationsMetric.ts
import { MetricService } from "../services/metric.service";
import { Metric } from "./Metric";
import { SpeedSensorServiceEvent } from './SpeedSensorServiceEvent';

export class WheelRotationsMetric extends Metric {
    private speedSensorHandler = this.speedSensorEventHandler.bind(this);

    constructor(metricService: MetricService) {
        super('Wheel Rotations', '', metricService, '');
    }

    startLogging(): void {
        this.metricService.getSpeedSensorService().subscribeForSpeedData(this.speedSensorHandler);
    }

    stopLogging(): void {
        this.metricService.getSpeedSensorService().unsubscribeForSpeedData(this.speedSensorHandler);
    }

    speedSensorEventHandler(event: SpeedSensorServiceEvent): void {
        if (!event.speedSensorData.cumulativeWheelRevolutions) {
            return;
        }

        this.addValue(event.speedSensorData.cumulativeWheelRevolutions, event.speedSensorData.timestamp);
    }
}

File: android/app/src/main/java/com/openbikecomputer/model/WheelRotationsMetric.java
package com.openbikecomputer.model;

import com.openbikecomputer.service.IMetricService;
import com.openbikecomputer.service.OnSpeedData;

public class WheelRotationsMetric extends Metric implements OnSpeedData {
    public WheelRotationsMetric(IMetricService metricService) {
        super(MetricType.WHEEL_ROTATIONS, "Wheel Rotations", "", metricService, "");
    }

    @Override
    public void startLogging() {
        metricService.getSpeedSensorService().subscribeForSpeedData(this);
    }

    @Override
    public void stopLogging() {
        metricService.getSpeedSensorService().unsubscribeForSpeedData(this);
    }

    @Override
    public void onSpeedData(SpeedSensorData data) {
        if (data.cumulativeWheelRevolutions == null || data.cumulativeWheelRevolutions == 0) {
            return;
        }
        addValue(data.cumulativeWheelRevolutions, data.timestamp);
    }
}
